c2corg_api/ext/colander_ext.py

```diff
@@ -53,7 +53,10 @@ def deserialize(self, node, cstruct):
         if cstruct is null or cstruct == '':
             return null
         try:
-            return wkbelement_from_geojson(cstruct, self.srid)
+            geom_type = json.loads(cstruct)['type'].upper()
+            allowed_types = node.typ.geometry_types
+            if geom_type in allowed_types:
```

Comment:
Could you raise an `Invalid` exception if it is not a valid type? Something like "Expected: ... Got: ...".